tests/integration/ambient/baseline_test.go

```diff
@@ -2426,3 +2517,28 @@ func TestDirect(t *testing.T) {
 		})
 	})
 }
+
+func deleteWaypoints(t framework.TestContext, nsConfig namespace.Instance, sa string) {
```

Comment:
I'm not an expert on the framework here but noting that deleting waypoints here seems to cause other tests to fail, we might be causing side effects by doing that.

Worth chasing down the side effects (bug or test bug either way).

This func `deleteWaypoints` probably belongs in `pkg/test/framework/components/ambient/waypoint.go`?

and should probably either be a func on the WaypointProxy interface, or take an instance of WaypointProxy.

If other tests are still failing with the waypoint delete, it might be because they expect one to be in place but do not explicitly assert/create one.